main/FatesHistoryInterfaceMod.F90

```diff
@@ -1511,7 +1518,12 @@ subroutine update_history_dyn(this,nc,nsites,sites)
                hio_ddbh_canopy_si_scag              => this%hvars(ih_ddbh_canopy_si_scag)%r82d, &
                hio_ddbh_understory_si_scag          => this%hvars(ih_ddbh_understory_si_scag)%r82d, &
                hio_mortality_canopy_si_scag         => this%hvars(ih_mortality_canopy_si_scag)%r82d, &
-               hio_mortality_understory_si_scag     => this%hvars(ih_mortality_understory_si_scag)%r82d)
+               hio_mortality_understory_si_scag     => this%hvars(ih_mortality_understory_si_scag)%r82d, &
+               hio_h2oveg_dead_si                   => this%hvars(ih_h2oveg_dead_si)%r81d, &
```

Comment:
@xuchongang, I'm running tests on cheyenne and running into problems at this line.
The problem is that these diagnostics are only allocated if hydraulics is turned on, yet this routine wants to associate a pointer to them even when hydraulics is off, and it is returning errors.
However, I'm wondering, do we even want these diagnostics calculated in this subroutine? This is only called once per day, but my impression was that these water contents are updated sub-daily right? Shouldn't we move these calculations to update_history_hydraulics anyway?

Reply:
After looking at this again, it looks like these should be updated sub-daily:

ih_h2oveg_si
ih_h2oveg_hydro_err_si

And these should be updated daily:
ih_h2oveg_dead_si
ih_h2oveg_recruit_si
ih_h2oveg_growturn_err_si
ih_h2oveg_pheno_err_si

Seem right to you @xuchongang @bchristo ?

Reply:
@rgknox yes this is correct. thanks!